src/libraries/System.Drawing.Primitives/src/System/Drawing/KnownColorTable.cs

```diff
@@ -495,14 +499,43 @@ public static uint GetSystemColorArgb(KnownColor color)
         {
             Debug.Assert(Color.IsKnownColorSystem(color));
 
-            return ColorTranslator.COLORREFToARGB(Interop.User32.GetSysColor((byte)ColorValueTable[(int)color]));
+            if (!SystemColors.s_useAlternativeColorSet || HighContrastEnabled())
+            {
+                return ColorTranslator.COLORREFToARGB(Interop.User32.GetSysColor((byte)ColorValueTable[(int)color]));
+            }
+
+            int index = color <= KnownColor.WindowText ? (int)color : (int)color - (int)KnownColor.ButtonFace + (int)KnownColor.WindowText + 1;
+            return AlternateSystemColors[index];
+        }
+
+        private static unsafe bool HighContrastEnabled()
+        {
+            Interop.User32.HIGHCONTRASTW highContrast = default;
+
+            // Note that the documentation for HIGHCONTRASTW says that the lpszDefaultScheme member needs to be
+            // freed, but this is incorrect. No internal users ever free the pointer and the pointer never changes.
+            highContrast.cbSize = (uint)sizeof(Interop.User32.HIGHCONTRASTW);
+            bool success = Interop.User32.SystemParametersInfoW(
+                Interop.User32.SystemParametersAction.SPI_GETHIGHCONTRAST,
+                highContrast.cbSize,
+                &highContrast,
+                0); // This has no meaning when getting values
+
+            return success && highContrast.dwFlags.HasFlag(Interop.User32.HIGHCONTRASTW_FLAGS.HCF_HIGHCONTRASTON);
         }
 #else
+
         public static uint GetSystemColorArgb(KnownColor color)
         {
             Debug.Assert(Color.IsKnownColorSystem(color));
 
-            return ColorValueTable[(int)color];
+            if (!SystemColors.s_useAlternativeColorSet)
```

Comment:
Why no high-contrast check here, but you did above?

Reply:
Nothing to check on Unix.